### PR TITLE
Adds the ability to set the offset of the note timestamps to the begi…

### PR DESCRIPTION
**Pull request by kmalta**
…nning of the recording, instead of the offset of the first note received.

### CODE REVIEW

File: music/src/core/recorder.ts [hunk 1]
```diff
@@ -27,11 +27,14 @@ import {DEFAULT_QUARTERS_PER_MINUTE} from './constants';
  * @param playClick Whether to play a click track while recording.
  * @param playCountIn Whether to play a count-in click at the beginning of
  * the recording.
+ * @param recordStartTimeOffset Whether to start the note time offset at
```
[the remark this answers is not in the document]
Comment (adarob) [reply music/src/core/recorder.ts]: I'm a bit confused about how this currently works. Say we have a click track and then someone plays their first note on the 2. Does the 2 then become the 1 in the recorded output?

[the remark this answers is not in the document]
Comment (notwaldorf) [reply music/src/core/recorder.ts]: Currently you press record and wait a while, the first note becomes offset to be at time 0, yeah (this was more important before we added the click track, and we never updated it for the click track).

[the remark this answers is not in the document]
Comment (adarob) [reply music/src/core/recorder.ts]: my preference would be that we have something like startRecordAtFirstNote that defaults to false...

[the remark this answers is not in the document]
Comment (notwaldorf) [reply music/src/core/recorder.ts]: 👍 with me!

File: music/src/core/recorder.ts [hunk 2]
```diff
@@ -27,14 +27,15 @@ import {DEFAULT_QUARTERS_PER_MINUTE} from './constants';
  * @param playClick Whether to play a click track while recording.
  * @param playCountIn Whether to play a count-in click at the beginning of
  * the recording.
- * @param recordStartTimeOffset Whether to start the note time offset at
- * the start of the recording.  Defaults to when the first note is received.
+ * @param startRecordAtFirstNote Whether to start the note time offset at
+ * the start of the recording or at the first note received.  Defaults to
```
[the remark this answers is not in the document]
Comment (notwaldorf) [reply music/src/core/recorder.ts]: also "startRecording" in the variable name. It's already long enough and letters are free.

File: music/src/core/recorder.ts [hunk 3]
```diff
@@ -268,6 +269,22 @@ export class Recorder {
     });
   }
 
+  /**
+   * Clears the NoteSequence and stops the recording.
+   * @returns a non-quantized `NoteSequence` containing all the currently
+   * recorded notes.
+   */
+  clearNoteSequence(): NoteSequence {
```
[the remark this answers is not in the document]
Comment (notwaldorf) [reply music/src/core/recorder.ts]: reset() sg

[the remark this answers is not in the document]
Comment (notwaldorf) [reply music/src/core/recorder.ts]: and in the comment, "clears the previously recorded NoteSequence..." etc.

[the remark this answers is not in the document]
Comment (notwaldorf) [reply music/src/core/recorder.ts]: and shouldn't return anything right? It should just clear and be done with.

[the remark this answers is not in the document]
Comment (kmalta) [reply music/src/core/recorder.ts]: It should still stop() though, right?

[the remark this answers is not in the document]
Comment (adarob) [reply music/src/core/recorder.ts]: i think it's okay to return in case people just want a one-stop-shot

[the remark this answers is not in the document]
Comment (notwaldorf) [reply music/src/core/recorder.ts]: but you're getting back an empty note sequence. what are you gonna do with it?

[the remark this answers is not in the document]
Comment (adarob) [reply music/src/core/recorder.ts]: why is it empty?

File: music/src/core/recorder.ts [hunk 4]
```diff
@@ -59,6 +63,7 @@ export class Recorder {
   private notes: NoteSequence.Note[] = [];
   private onNotes: Map<number, NoteSequence.Note>;
   private midiInputs: WebMidi.MIDIInput[] = [];
+  private startRecordAtFirstNote: boolean;
```
Comment (notwaldorf) [inline music/src/core/recorder.ts]: argh, i thought i sent this already, but can this be "startRecordingAtFirstNote", please? It sounds a bit more human. That's my last comment, pinky swear i LGTM after

**Pull request merged**